### ISSUE TITLE
pe::write - support for partial filling of section contents

### ISSUE THREAD

**Issue by osiewicz** [gimli-rs/object#396]
Hi,
Thanks for the work you put into this library - it is really great and convenient. I use it in my hobby PE linker project.

As far as I can tell, the only way to fill a PE section is via a `object::write::pe::Writer::write_section`, and it's kind of all-or-nothing in nature (a given offset can be filled at most once, as later on the alignment kicks in). See [the definition](https://github.com/gimli-rs/object/blob/master/src/write/pe.rs#L513). I understand why this is an intended behaviour. 
Is there a way to improve upon it? My gripe with it is that I need to provide *one* slice with full contents of the section - which requires unnecessary copying when the contents of final section are scattered over several .obj files.

I'd be happy to submit a PR that adresses the issue. 

TL;DR: is it possible to achieve:
```
let writer: pe::Writer = ...;
// Reserve everything for writer..
let foo: [u8; 16] = [0; 16];
let bar: [u8; 24] = [0; 24];
let range = writer.reserve_section(".text", some_flags, 40, 40)
writer.write_section(range.file_offset, &foo);
writer.write_section(range.file_offset + foo.size(), &bar);
```
Instead of:
```
let writer: pe::Writer = ...; 
// Reserve everything for writer..
let foo: [u8; 16] = [0; 16];
let bar: [u8; 24] = [0; 24];
let range = writer.reserve_section(".text", some_flags, 40, 40)
let mut merged = vec![];
merged.extend_from_slice(foo);
merged.extend_from_slice(bar);
writer.write_section(range.file_offset, merged);
```
or have some kind of alternative API to do that?

**Comment by philipc** [comment]
You don't have to call `write_section` to write the section data. It is just a convenience function. So instead you can call `pad_until` to get to the start of the section data, followed by `write` as many times as you want, followed by `write_align` to add the trailing padding.

**Comment by philipc** [comment]
Adding a new convenience function to do the trailing padding might be useful though, since we don't expose the file alignment anywhere.

**Comment by osiewicz** [comment]
Thanks! I've totally missed that - the solution you've provided will do the trick for me.

The issue can be closed now, unless you want to track the padding functionality with it.

**Comment by osiewicz** [comment]
> Adding a new convenience function to do the trailing padding might be useful though, since we don't expose the file alignment anywhere.

Would [this patch](https://gist.github.com/osiewicz/351e1bd225b56b3e530d69fda809eb51) do the trick? If so, I'll create a PR.

**Comment by philipc** [comment]
Yes, but let's call it `write_file_align` for consistency with the existing `write_align`, and place it after `write_align`.
